pkg/controller/jobframework/setup_test.go

```diff
@@ -135,18 +156,42 @@ func TestSetupControllers(t *testing.T) {
 				t.Fatalf("Failed to setup manager: %v", err)
 			}
 
-			gotError := manager.setupControllers(mgr, logger, tc.opts...)
+			gotError := manager.setupControllers(context.Background(), mgr, logger, time.Millisecond * 20, tc.opts...)
 			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); len(diff) != 0 {
 				t.Errorf("Unexpected error from SetupControllers (-want,+got):\n%s", diff)
 			}
 
-			if diff := cmp.Diff(tc.wantEnabledIntegrations, manager.enabledIntegrations.SortedList()); len(diff) != 0 {
+			if slices.Contains(tc.wantEnabledIntegrations, "ray.io/raycluster") {
+				testDelayedIntegration(mgr, manager)
+			}
+
+			manager.mu.RLock()
+			diff := cmp.Diff(tc.wantEnabledIntegrations, manager.enabledIntegrations.SortedList())
+			manager.mu.RUnlock()
```

Comment:
I think it's better to create a method `getEnabledIntegrations()` that will lock `enabledIntegrations` and return copy of value. You can use it where it's need.